erpnext/stock/get_item_details.py

```diff
@@ -69,8 +69,70 @@ def get_item_details(args):
 	if args.get("is_subcontracted") == "Yes":
 		out.bom = get_default_bom(args.item_code)
 
+	if args.document_type in ["Quotation Item", "Sales Order Item"]:
+		# calculate rate by appling discount to total_margin
+		if(args.total_margin):
+			out["rate"] = discount_on_total_margin(args.total_margin, out.discount_percentage);
+		elif not args.type:
+			data = set_margin_fields(out.pricing_rule, out.price_list_rate)
+			out["rate"] = discount_on_total_margin(data.get("total_margin"), out.discount_percentage)
+			out.update(data)
+
 	return out
```

Comment:
@mbauskar since this is related to pricing rule, the correct place to put this code is here:

https://github.com/frappe/erpnext/blob/develop/erpnext/accounts/doctype/pricing_rule/pricing_rule.py#L158

Usually its easy to put everything related to a feature (like pricing rule) in one place so that it will be easy to debug. If this is here, someone needs to remember that margin is in `get_item_details` and discount is in `apply_pricing_rule`

Also since discount and margin are similar, they should be together.

Can you fix. Thanks!